File: data/CodeLlama/readable_format_HPEC/cpp/file142.cpp
#include <iostream>
#include <omp.h>

// Function prototype
int fib(int n);

int main() {
    int input = 30;
    int result;

    #pragma omp parallel
    {
        #pragma omp single
        {
            result = fib(input);
        }
    }

    std::cout << "Fib for " << input << " = " << result << std::endl;

    return 0;
}

// Recursive function fib
int fib(int n) {
    int i, j;

    if (n < 2) {
        return n;
    } else {
        #pragma omp task shared(i)
        i = fib(n - 1);
        #pragma omp task shared(j)
        j = fib(n - 2);
        #pragma omp taskwait
        return i + j;
    }
}

File: data/CodeLlama/readable_format_HPEC/fortran/file142.f90
module DRB106
    implicit none
    integer (kind=4) input

contains
    recursive function fib(n) result(r)
        use omp_lib
        implicit none
        integer (kind=4) :: n, i, j, r

        if (n<2) then
            r = n
        else
            !$omp task shared(i)
            i = fib(n-1)
            !$omp end task
            !$omp task shared(j)
            j = fib(n-2)
            !$omp end task
            r = i+j
        end if
        !$omp taskwait
    end function
end module

program DRB106_taskwaitmissing_orig_yes
    use omp_lib
    use DRB106
    implicit none

    integer :: result
    input = 30

    !$omp parallel
    !$omp single
    result = fib(input)
    !$omp end single
    !$omp end parallel

    print*,'Fib for ',input,' =',result

end program
